appinventor/appengine/src/com/google/appinventor/client/boxes/SourceStructureBox.java

```diff
@@ -52,4 +54,13 @@ private SourceStructureBox() {
   public SourceStructureExplorer getSourceStructureExplorer() {
     return sourceStructureExplorer;
   }
+
+  public void show(YaFormEditor form) {
```

Comment:
It seems like it would make more sense for this to take a MockForm rather than a YaFormEditor. This actually would work better for different layouts because the combined layout may not make much use of YaFormEditor, and the structure being rendered is actually the structure from the MockForm, it just happens that today YaFormEditor owns it.